### PR TITLE
add support for both h5 and n4 CYGNSS files

### LINKED ISSUE #1465: CYGNSS wind files not in hdf5 format netcdf4 only
## Description

the most recent CYGNSS data is not following hdf5 format but is decodable using the netCDF4 library

## Requirements

h5py and netCDF4 python libraries

## Acceptance Criteria (Definition of Done)
decodes both hdf5 supported and netCDF4 only supported CYGNSS files

### PR DESCRIPTION
**Pull request by BenjaminRuston**
## Description

support CYGNSS file in both h5 and netcdf4 formats

## Issue(s) addressed

Resolves #1465 

## Checklist

- [ ] I have performed a self-review of my own code

### CODE REVIEW

File: src/hdf5/osw_2ioda.py [hunk 1]
```diff
@@ -187,7 +193,13 @@ def main(args):
     writer.BuildIoda(ioda_data, VarDims, varAttrs, GlobalAttrs)
 
 
-def get_data_source(afile):
+def get_data_source(afile, nc4=False):
+    if nc4:
+        if  'title' in afile.ncattrs() and 'CYGNSS' in afile.getncattr('title'):
+            return 'CYGNSS'
+        else:
+            return 'unsupported file type'
```
Comment (BenjaminRuston) [inline src/hdf5/osw_2ioda.py:201]: should probably have the decoder fall over if this `unsupported` is returned

**Comment by smaticka** [comment]
@BenjaminRuston, as discussed, rather than making all of these additions, we could instead run this `os` command to make the files compatible with the current converter. 

`nccopy -k 'netCDF-4' cyg.ddmi.s20240109-000001-e20240109-235959.l2.wind_trackgridsize25km_NOAAv1.2_L1a21.d21.nc cyg_copy.nc`

@ashley314. Wanted to keep you in the loop on this converter change (which we won't do). Essentially the newer `cygnss` files are not compatible with hd5f python package, but the os/terminal command above will convert the files to then be used by the current converter as is. Ben said this `nccopy -k 'netCDF-4'` will belong in the ingest script, so wanted you to be aware that this will be needed as a check. Some sort of check similar to lines 100-103 that Ben added might be needed and then as an `else` option, using `nccopy` to convert. (`nccopy -k` can also check the filetype I believe). etc etc :)

**Comment by smaticka** [comment]
@BenjaminRuston can we close this now?

**Comment by ashley314** [comment]
Thanks @smaticka for the information. I'll make note of it for when we add these obs to the suite

**Comment by BenjaminRuston** [comment]
yes thanks @smaticka agree this should be done on the ingest step, and will discuss with the data provider if they can't remedy this directly on production

**Pull request closed without merging**